### PR TITLE
Refactor the backend impl package

### PR DESCRIPTION
**Pull request by mkondratek**
Basicly extract some private static classes to separate files. cleaner. easier to test (having https://github.com/VirtusLab/git-machete-intellij-plugin/issues/1365 in mind)

### CODE REVIEW

File: backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/repository/GitMacheteRepository.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import com.virtuslab.gitmachete.backend.api.ILocalBranchReference;
+import com.virtuslab.gitmachete.backend.impl.aux.Aux;
+import com.virtuslab.gitmachete.backend.impl.aux.CreateGitMacheteRepositoryAux;
+import com.virtuslab.gitmachete.backend.impl.aux.DiscoverGitMacheteRepositoryAux;
+import com.virtuslab.gitmachete.backend.impl.hooks.PreRebaseHookExecutor;
+import com.virtuslab.gitmachete.backend.impl.hooks.StatusBranchHookExecutor;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@RequiredArgsConstructor
+public class GitMacheteRepository implements IGitMacheteRepository {
```
Comment (mkondratek) [inline backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/repository/GitMacheteRepository.java:24]: I dropped `@CustomLog` here. 
however there are some `@Loggable` below in this file.
`LOG` calls were from Aux methods only.

is it ok?

Comment (PawelLipski) [reply backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/repository/GitMacheteRepository.java:24]: Apparently [`@Loggable`](https://aspects.jcabi.com/apidocs-0.22.6/com/jcabi/aspects/Loggable.html) takes care of providing a correct SLF4J logger... so it seems it's okay

File: backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepositoryCache.java [hunk 2]
```diff
@@ -18,6 +18,7 @@
 import com.virtuslab.gitmachete.backend.api.IGitMacheteRepositoryCache;
 import com.virtuslab.gitmachete.backend.impl.hooks.PreRebaseHookExecutor;
 import com.virtuslab.gitmachete.backend.impl.hooks.StatusBranchHookExecutor;
+import com.virtuslab.gitmachete.backend.impl.repository.GitMacheteRepository;
 
 public class GitMacheteRepositoryCache implements IGitMacheteRepositoryCache {
```
Comment (PawelLipski) [inline backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepositoryCache.java:23]: Should this class go to `repository` subpackage? Or maybe the `repository` subpackage isn't really needed?

File: backend/impl/src/test/java/com/virtuslab/gitmachete/backend/unit/BaseGitMacheteRepositoryUnitTestSuite.java [hunk 3]
```diff
@@ -41,8 +32,6 @@ protected Object aux(IGitCoreBranchSnapshot... localCoreBranches) {
     // cannot be mocked as it is final
     val statusBranchHookExecutor = StatusBranchHookExecutor.of(gitCoreRepository);
 
-    return Whitebox
-        .getConstructor(AUX_CLASS, IGitCoreRepository.class, StatusBranchHookExecutor.class, PreRebaseHookExecutor.class)
-        .newInstance(gitCoreRepository, statusBranchHookExecutor, /* preRebaseHookExecutor */ null);
+    return new CreateGitMacheteRepositoryAux(gitCoreRepository, statusBranchHookExecutor, /* preRebaseHookExecutor */ null);
```
Comment (PawelLipski) [inline backend/impl/src/test/java/com/virtuslab/gitmachete/backend/unit/BaseGitMacheteRepositoryUnitTestSuite.java:35]: <3

**Review by PawelLipski — APPROVED** [review]

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**